packages/ui/src/components/Visualization/Custom/ItemDeleteStepWithConfirm.tsx

```diff
@@ -0,0 +1,32 @@
+import { TrashIcon } from '@patternfly/react-icons';
+import { ContextMenuItem } from '@patternfly/react-topology';
+import { FunctionComponent, PropsWithChildren, useCallback, useContext } from 'react';
+import { IDataTestID } from '../../../models';
+import { IVisualizationNode } from '../../../models/visualization/base-visual-entity';
+import { EntitiesContext } from '../../../providers/entities.provider';
+import { DeleteModalContext } from '../../../providers/delete-modal.provider';
+
+interface ItemDeleteStepWithConfirmProps extends PropsWithChildren<IDataTestID> {
+  vizNode: IVisualizationNode;
```

Comment:
What if we pass a boolean property, perhaps called `shouldConfirmBeforeDeletion`, with a default value of `true`, and then we use that property in the component to check whether we should ask for confirmation or not?

The advantage of this will be that we don't need to create an additional component just for the asking confirmation, we can use the same component in both cases.